paddle/fluid/operators/conv_cudnn_helper.h

```diff
@@ -251,7 +251,7 @@ struct SearchAlgorithm<cudnnConvolutionFwdAlgoPerf_t> {
               args.cdesc.desc(), args.odesc.desc(), kNUM_CUDNN_FWD_ALGS,
               &perf_count, perf_results.get()));
       algo = (perf_results.get())[best_algo_idx].algo;
-      workspace_size = GetWorkspaceSize(args, algo);
```

Comment:
The test code about whether the return value of `GetWorkspaceSize` interface is equal with `(perf_results.get)[0].memory` is below :
```cpp 
    int perf_count;
    int best_algo_idx = 0;
    std::unique_ptr<perf_t[]> perf_results(new perf_t[kNUM_CUDNN_FWD_ALGS]);
    PADDLE_ENFORCE_GPU_SUCCESS(
        platform::dynload::cudnnGetConvolutionForwardAlgorithm_v7(
            args.handle, args.idesc.desc(), args.wdesc.desc(),
            args.cdesc.desc(), args.odesc.desc(), kNUM_CUDNN_FWD_ALGS,
            &perf_count, perf_results.get()));
    algo = (perf_results.get())[best_algo_idx].algo;
    workspace_size = GetWorkspaceSize(args, algo);

    if (workspace_size == (perf_results.get())[0].memory) {
       VLOG(6) << "workspace_size == perf_results.get())[0].memory" ;
   }
```